src/test/main.cpp

```diff
@@ -24,3 +24,17 @@ const std::function<void(const std::string&)> G_TEST_LOG_FUN = [](const std::str
     if (!should_log) return;
     std::cout << s;
 };
+
+/**
+ * Retrieve the command line arguments from boost.
+ * Allows usage like:
+ * `test_bitcoin --run_test="net_tests/cnode_listen_port" -- -checkaddrman=1 -printtoconsole=1`
+ * which would return `["-checkaddrman=1", "-printtoconsole=1"]`.
+ */
+const std::function<std::vector<const char*>()> G_TEST_COMMAND_LINE_ARGUMENTS = []() {
```

Comment:
c79eb4ad9f0adf733a9da59d87dc3a0d10a90738 Does it makes sense for `G_TEST_COMMAND_LINE_ARGUMENTS` to be named as a constant (as it should update at runtime, yes?)

Reply:
Good question. `G_TEST_COMMAND_LINE_ARGUMENTS` is a constant (its value is the function itself, we don't change it to another function). So, strictly speaking this is a `const` variable. But we use it as a function - `G_TEST_COMMAND_LINE_ARGUMENTS()`.  I used that name for consistency with the already existent `G_TEST_LOG_FUN` which is defined a few lines earlier.

Semantically it is a function and personally I think it would be most logical to name it like a function `TestCommandLineArguments()`. Like I did with `CreateSock`:

https://github.com/bitcoin/bitcoin/blob/77a2f5d30c5ecb764b8a7c098492e1f5cdec90f0/src/netbase.h#L190

Maybe the guidelines should be updated to clarify how to name things that are both variables and functions.

Reply:
Upper case might be correct, see #18568. Though, maybe the `G_` should be removed, as upper case implies `G_`?

Reply:
Anyway, for this PR I think consistency with the neighboring `G_TEST_LOG_FUN` is the most important. Marking this as resolved, feel free to comment further / unresolve it.